packages/runtime/container-runtime/src/containerRuntime.ts

```diff
@@ -2615,29 +2609,25 @@ export class ContainerRuntime
 		}
 	}
 
-	private async loadIdCompressor(): Promise<void | undefined> {
+	private loadIdCompressor(): void {
 		if (
 			this._idCompressor === undefined &&
-			this.sessionSchema.idCompressorMode !== undefined &&
-			this._loadIdCompressor === undefined
+			this.sessionSchema.idCompressorMode !== undefined
 		) {
-			this._loadIdCompressor = this.createIdCompressor()
-				.then((compressor) => {
-					// Finalize any ranges we received while the compressor was turned off.
-					const ops = this.pendingIdCompressorOps;
-					this.pendingIdCompressorOps = [];
-					for (const range of ops) {
-						compressor.finalizeCreationRange(range);
-					}
-					assert(this.pendingIdCompressorOps.length === 0, 0x976 /* No new ops added */);
-					this._idCompressor = compressor;
-				})
-				.catch((error) => {
-					this.mc.logger.sendErrorEvent({ eventName: "IdCompressorDelayedLoad" }, error);
-					throw error;
-				});
+			try {
+				this._idCompressor = this.createIdCompressorFn();
```

Comment:
Do we expect there to be errors coming from this call now that it is no longer async? I believe there could be issues with serialization/deserialization (theoretically...this stuff is really well tested and has never had an issue) but is there a particular reason we want to log those errors as being related to delayed loading of the ID compressor as in the catch block? Should we remove the try/catch? Or at least change the telemetry event name?

Reply:
Yeah, I just kept it as it was there before. I also don't expect it so removing.